samples/snippets/src/main/java/com/example/bigquerystorage/WriteToDefaultStream.java

```diff
@@ -135,22 +135,29 @@ private static class DataWriter {
     // Track the number of in-flight requests to wait for all responses before shutting down.
     private final Phaser inflightRequestCount = new Phaser(1);
     private final Object lock = new Object();
+
     private JsonStreamWriter streamWriter;
 
     @GuardedBy("lock")
     private RuntimeException error = null;
 
     public void initialize(TableName parentTable)
         throws DescriptorValidationException, IOException, InterruptedException {
+      initialize(parentTable.toString());
+    }
+
+    private void initialize(String streamOrTableName)
+        throws DescriptorValidationException, IOException, InterruptedException {
       // Use the JSON stream writer to send records in JSON format. Specify the table name to write
       // to the default stream.
       // For more information about JsonStreamWriter, see:
       // https://googleapis.dev/java/google-cloud-bigquerystorage/latest/com/google/cloud/bigquery/storage/v1/JsonStreamWriter.html
       streamWriter =
```

Comment:
we don't need to call close() on the previous streamwriter?

Reply:
Yifat asked the same question, aka, stream worker will call close automatically on GC. 
https://github.com/googleapis/java-bigquerystorage/blob/main/google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java#L445

But I am also not against add one explicitly. WDYT?

Reply:
GC will trigger much time later than this overwrite and GC won't care autoclosable (it won't trigger close() during GC)
https://stackoverflow.com/questions/8321767/closeable-in-garbage-collection

Autoclosable is explicitly used for try with resource which is not related to GC 
Let's close the previous one explicitly

Reply:
Good point!